modules/_labs/select/react/lib/Select.tsx

```diff
@@ -0,0 +1,681 @@
+      matchIndex = this.getIndexByStartString(0, this.keysSoFar, start);
+    }
+
+    // A match was found...
+    if (matchIndex > -1) {
+      // If the menu is hidden, immediately select the matched option
+      if (this.state.isMenuHidden) {
+        this.setState({
+          focusedOptionIndex: matchIndex,
+          label: this.optionLabels[matchIndex],
+          selectedOptionIndex: matchIndex,
+        });
+        this.updateInput(this.optionValues[matchIndex]);
+
+        // Otherwise (the menu is visible), simply focus the matched option
+      } else {
+        this.setState({focusedOptionIndex: matchIndex});
+      }
+    }
+  };
+
+  private startClearKeysSoFarTimer = () => {
+    if (this.clearKeysSoFarTimer) {
+      clearTimeout(this.clearKeysSoFarTimer);
+    }
+    this.clearKeysSoFarTimer = setTimeout(() => {
+      this.keysSoFar = '';
+    }, this.clearKeysSoFarTimeout);
+  };
+
+  constructor(props: SelectProps) {
+    super(props);
+    this.setOptionLookups();
+  }
+
+  componentDidMount() {
+    const {value} = this.props;
+
+    // If value exists, set state to that value...
+    if (value) {
+      const childIndex = this.getIndexByValue(value);
+
+      if (childIndex !== -1) {
+        this.setState({
+          label: this.optionLabels[childIndex],
+          selectedOptionIndex: childIndex,
+        });
+        return;
+      }
+    }
+
+    // ... otherwise, set state to the first option
+    this.setState({
+      label: this.optionLabels[0],
+    });
+  }
+
+  componentDidUpdate(prevProps: SelectProps, prevState: SelectState) {
+    const {children} = this.props;
+    const {isMenuHidden, focusedOptionIndex} = this.state;
+
+    if (children !== prevProps.children) {
+      this.setOptionLookups();
+    }
+
+    // If the menu was just displayed, scroll the focused option into
+    // center view
+    if (!isMenuHidden && prevState.isMenuHidden) {
+      this.scrollFocusedOptionIntoView(true);
+
+      // Otherwise, if the menu is displayed AND the focused option changed
+      // since the last render, scroll the focused option into view, but
+      // do NOT center it
+    } else if (!isMenuHidden && focusedOptionIndex !== prevState.focusedOptionIndex) {
+      this.scrollFocusedOptionIntoView(false);
+    }
+  }
+
+  componentWillUnmount() {
+    // Clear timers
+    if (this.removeMenuTimer) {
+      clearTimeout(this.removeMenuTimer);
+    }
+    if (this.clearKeysSoFarTimer) {
+      clearTimeout(this.clearKeysSoFarTimer);
+    }
+  }
+
+  handleSelectClick = (event: React.MouseEvent<HTMLButtonElement>): void => {
+    // Cancel default handling of the event to ensure we maintain
+    // control of focus
+    event.preventDefault();
+
+    this.toggleMenu(this.state.isMenuHidden);
+  };
+
+  handleSelectMouseDown = (event: React.MouseEvent<HTMLButtonElement>): void => {
+    // Cancel default handling of the event to ensure we maintain
+    // control of focus
+    event.preventDefault();
```

Comment:
Is this needed for both click and mousedown?

Reply:
Preventing default is only needed on mouse down since that's the default action of a `mousedown` event. It shouldn't also be done on click

Reply:
Removed `preventDefault` from click handler.